### ISSUE TITLE
Some icons showing blank in context menu

### ISSUE THREAD

**Issue by Aetherinox** [FlorianWoelki/obsidian-iconize#309]
**Describe the bug**
Certain icons in the suggestion context menu appear blank. However, if you select one of the blank spots, the actual name of the icon appears in the note, but also doesn't show in either mode for Obsidian (Editing and Reading Mode). It just shows as the text to call the icon.

**To Reproduce**
1. Install Font Awesome 6 icons
2. Start typing out the name of an icon

**Expected behavior**
Shows icons

**Screenshots**
This example is of me typing out the starting name of icons with `:check` to see what all matches I can find. Eventually no actual icons show and it's all blank boxes.

![NzTLsFp](https://github.com/FlorianWoelki/obsidian-iconize/assets/118329232/174f348d-9c9f-4514-89d0-a9c9c9b77bcf)

<br />

Here is a list of the icon packages I have installed:
![LpUtGJ0](https://github.com/FlorianWoelki/obsidian-iconize/assets/118329232/6b47d813-38aa-4542-9e6b-0761e6f3fe85)

### Note
I do own and have Font-Awesome 6 Pro icons installed, but they should behave the same as the free ones or Font-Awesome 5.
As shown in the 2nd screenshot of my icon list, I keep the conformity that Font-Awesome normally uses which is
`far`: Font Awesome Regular
`fab`: Font Awesome Brands
`fas`: Font Awesome Solid
`fad`: Font Awesome Dualtone

<br />

Here is a small sample list of the icons that don't work
```
:FaBuildingCircleCheck:
:FaAnchorCircleCheck:
:FaBadgeCheck:
:FaBinaryCircleCheck:
:FaBoxCircleCheck:
:FaBuildingCircleCheck:
:FaCartCircleCheck:
:FaCheckDouble:
:FaCircleCheck:
:FaDog:
:FaDogLeashed:
:FaHotdog:
:FaDogLeashed:
:FaDogLeashed:
```

Seem to all start with `Fa`.

**Comment by FlorianWoelki** [comment]
Thank you for reporting this issue. I am currently using Font-Awesome-Solid from the premade custom icon packs (you can find these under `Browse icon packs`) and it works perfectly without any issues.

Do you have some kind of reproduction vault, so that I can reproduce this issue more easily?

**Comment by Aetherinox** [comment]
I'm running some tests at the moment for the "blank" icons.
But I've also started a new vault, and all I've installed are two fontawesome 6 icon packs, fas, and fad

![ZtDX3Fd](https://github.com/FlorianWoelki/obsidian-iconize/assets/118329232/5955f499-da95-4a28-aa9b-f00dfc99d932)


It shows the few icons I added, but for fas and fad, it shows the same icons.

![nqojthB](https://github.com/FlorianWoelki/obsidian-iconize/assets/118329232/60f1f899-81e6-4288-a312-f9c63c025e95)

I added all 4 of the icons in the list, and they appear to have the same names

```
:FaDog:
:FaDogLeashed:
:FaDog:
:FaDogLeashed:
```

It's not doing something like FasDog or FadDog. It's sticking to FaDog, and two icons from two collections collide.

**Comment by Aetherinox** [comment]
So that's the problem. No two icon packs can contain the same first two letters, otherwise they'll start overriding each other.

![OnE8rcH](https://github.com/FlorianWoelki/obsidian-iconize/assets/118329232/756be9b1-b3db-4744-8fb0-ae8a8b5aa1a2)

Once I uninstalled all the icons and re-added them using different abbreviations that didn't collide with others, it shows multiple icons fine.

![FgscVBh](https://github.com/FlorianWoelki/obsidian-iconize/assets/118329232/5bb5c4a0-0f10-4a67-be64-e68e0b1dbef6)

Going to make for a pain with trying to remember which is which, because a few of the FA Pro icons have similar style names.

**Comment by Aetherinox** [comment]
Yeah, it seems to be if you have icons of the same name, that also are in packs starting with the same two letters, it causes them to break, or one icon pack will take priority.


For FontAwesome, none of the icons are prefixed with anything. Just the name of the icon itself:

![vVSBvie](https://github.com/FlorianWoelki/obsidian-iconize/assets/118329232/a5740e7a-5d6c-4d80-9cfb-a00189bce774)
